### ISSUE TITLE
Rename EclipseState -> ReservoirState?

### ISSUE THREAD

**Issue by joakim-hove** [OPM/opm-simulators#2540]

**Comment by atgeirr** [comment]
I have also thought that EclipseState is becoming a rather odd name. If we implement other input formats/sources such as Intersect, python code, or what have you, it is better with a new name. But I do not really like ReservoirState. To me, that sounds like the place where I would find current pressure, saturations and similar variables. EclipseState is a state in the sense that the input deck "modifies a state machine", but from Flow's point of view it provides the case or configuration. So what about SimulatorConfiguration? Or SimulatorSetup? Or (too radical...) just Input?

**Comment by joakim-hove** [comment]
> So what about SimulatorConfiguration? Or SimulatorSetup? Or (too radical...) just Input?

How about `ResInput`? I would be willing to go with just `Input` also.

**Comment by bska** [comment]
I too am in favour of renaming `EclipseState` to Something Else[tm].  I'm not really keen on anything that uses the term `Input` though.  The `Schedule` is input too, and that isn't covered by the suggested renaming.

To me the current `EclipseState` is a static model description (properties, geometry, simulation graph, unit system &c) whereas the `Schedule` represents the dynamic inputs/constraints.  I hesitate to use the terms "static model" and "dynamic model" because those have specific connotations that we might not be fully able to capture.  Off the top of my head I'd suggest `StaticModelDescription`, but I haven't really tought too much about it.

**Comment by joakim-hove** [comment]
> The Schedule is input too, and that isn't covered by the suggested renaming.

We could rename that as well.

> Off the top of my head I'd suggest StaticModelDescription

Ohhh - that is loooong :-)

> I'm not really keen on anything that uses the term Input though. The Schedule is input too, and that isn't covered by the suggested renaming.

What about `StaticInput` and `DynamicInput`?

**Comment by atgeirr** [comment]
> What about `StaticInput` and `DynamicInput`?

Not bad. However I think `Schedule` is a good name, that also corresponds well to the SCHEDULE sections of input decks, that we should keep.

**Comment by joakim-hove** [comment]
> However I think Schedule is a good name

I agree that `Schedule` is a *much* better name than `EclipseState`!
